templates/docker-init.j2

```diff
@@ -1,12 +1,61 @@
 # Generated by Ansible for {{ansible_fqdn}}
 description "Docker daemon"
 
-start on filesystem
+start on (filesystem and net-device-up IFACE!=lo)
 stop on runlevel [!2345]
+limit nofile 524288 1048576
+limit nproc 524288 1048576
 
 respawn
 
+kill timeout 20
+
+pre-start script
+        # see also https://github.com/tianon/cgroupfs-mount/blob/master/cgroupfs-mount
+        if grep -v '^#' /etc/fstab | grep -q cgroup \
+                || [ ! -e /proc/cgroups ] \
+                || [ ! -d /sys/fs/cgroup ]; then
+                exit 0
+        fi
+        if ! mountpoint -q /sys/fs/cgroup; then
+                mount -t tmpfs -o uid=0,gid=0,mode=0755 cgroup /sys/fs/cgroup
+        fi
+        (
+                cd /sys/fs/cgroup
+                for sys in $(awk '!/^#/ { if ($4 == 1) print $1 }' /proc/cgroups); do
+                        mkdir -p $sys
+                        if ! mountpoint -q $sys; then
+                                if ! mount -n -t cgroup -o $sys cgroup $sys; then
+                                        rmdir $sys || true
+                                fi
+                        fi
+                done
+        )
+end script
+
 script
-  [ ! -f /etc/default/docker ] || . /etc/default/docker
-  /usr/bin/docker {{ docker_daemon_opts }} $DOCKER_OPTS
+        # modify these in /etc/default/$UPSTART_JOB (/etc/default/docker)
```

Comment:
Yes, this is originated from : https://github.com/docker/docker/blob/master/contrib/init/upstart/docker.conf